src/Telemetry/AzureDevOpsTimeline/AzureDevOpsTimeline.cs

```diff
@@ -176,15 +180,46 @@ private async Task RunLoop(CancellationToken cancellationToken)
                     build => azureServer.GetTimelineAsync(project, build.Id, cancellationToken)
                 );
 
+            // for each timeline, look for previousAttempts.timelineId timelines
+            // For each timeline, look at timeline.lastChangedOn for ingestion cutoff
+            // timeline.lastChangedOn will be before build.finishTime
+
             await Task.WhenAll(tasks.Select(s => s.Value));
+
+            // Look for retried timelines
+            List<(Build build, Task<Timeline> timelineTask)> retriedTimelineTasks = new List<(Build, Task<Timeline>)>();
+
+            foreach ((Build build, Task<Timeline> timelineTask) in tasks)
+            {
+                Timeline timeline = await timelineTask;
+
+                IEnumerable<string> additionalTimelineIds = timeline.Records
+                    .SelectMany(record => record.PreviousAttempts)
+                    .Where(attempt => !(attempt is null))
+                    .Select(attempt => attempt.TimelineId)
+                    .Distinct(StringComparer.OrdinalIgnoreCase);
```

Comment:
Is it wise to use "IgnoreCase" here? Are the timelineId's really case varying, and we are supposed to collapse them? It seems like it would probably be better to just use them as opaque strings from AzDO.

Reply:
I'm fine with opaque string. 

My concern was that the ID is actually a GUID, and different systems treat capitalization in GUIDs differently. I don't have any specific evidence of problems though.